static/profile/js/profile.js

```diff
@@ -0,0 +1,515 @@
+  }
+  
+  // Handling valid from date change
+  function dateChanged(event) {
+    var newdate = new Date(Nightscout.client.utils.mergeInputTime($('#pe_time').val(), $('#pe_date').val()));
+    if (mongoprofiles.length<2 || !mongoprofiles[1].startDate || mongoprofiles.length>=2 && new Date(mongoprofiles[1].startDate).getTime() === newdate.getTime()) {
+      $('#pe_submit').text('Update record').css('display','');
+      $('#pe_time').css({'background-color':'white'});
+      $('#pe_date').css({'background-color':'white'});
+      $('#pe_submit').css({'background-color':'buttonface'});
+    } else if (mongoprofiles.length<2 || new Date(mongoprofiles[1].startDate).getTime() < newdate.getTime()) {
+      $('#pe_submit').text('Create new record').css('display','');
+      $('#pe_time').css({'background-color':'green'});
+      $('#pe_date').css({'background-color':'green'});
+      $('#pe_submit').css({'background-color':'green'});
+    } else {
+      $('#pe_submit').css('display','none');
+      $('#pe_time').css({'background-color':'red'});
+      $('#pe_date').css({'background-color':'red'});
+      $('#pe_submit').css({'background-color':'red'});
+    }
+    if (event) {
+      event.preventDefault();
+    }
+  }
+  
+  // Handling html events and setting/getting values
+  function switchStyle(event) {
+    if (!$('#pe_perGIvalues').is(':checked')) {
+      $('#pe_simple').show('slow');
+      $('#pe_advanced').hide('slow');
+    } else {
+      $('#pe_simple').hide('slow');
+      $('#pe_advanced').show('slow');
+    }
+    if (event) {
+      event.preventDefault();
+    }
+  }
+  
+  function fillTimeRanges(event) {
+    var mmoltime = [
+      '0:00','0:30','1:00','1:30','2:00','2:30','3:00','3:30','4:00','4:30','5:00','5:30',
+      '6:00','6:30','7:00','7:30','8:00','8:30','9:00','9:30','10:00','10:30','11:00','11:30',
+      '12:00','12:30','13:00','13:30','14:00','14:30','15:00','15:30','16:00','16:30','17:00','17:30',
+      '18:00','18:30','19:00','19:30','20:00','20:30','21:00','21:30','22:00','22:30','23:00','23:30'
+    ];
+    var mgtime = [
+      '12:00AM','0:30AM','1:00AM','1:30AM','2:00AM','2:30AM','3:00AM','3:30AM','4:00AM','4:30AM','5:00AM','5:30AM',
+      '6:00AM','6:30AM','7:00AM','7:30AM','8:00AM','8:30AM','9:00AM','9:30AM','10:00AM','10:30AM','11:00AM','11:30AM',
+      '12:00PM','0:30PM','1:00PM','1:30PM','2:00PM','2:30PM','3:00PM','3:30PM','4:00PM','4:30PM','5:00PM','5:30PM',
+      '6:00PM','6:30PM','7:00PM','7:30PM','8:00PM','8:30PM','9:00PM','9:30PM','10:00PM','10:30PM','11:00PM','11:30PM'
+    ];
+    if (event) {
+      GUIToObject();
+    }
+    
+    function shouldAddTime(i,time,array) {
+      if (i===0 && time>0) { return false; }
+      if (i>0 && isNaN(toMinutesFromMidnight(c_profile[array][i-1].time))) { return false; }
+      if (i>0 && toMinutesFromMidnight(c_profile[array][i-1].time) >= time*30) { return false; } 
+      return true;
+    }
+ 
+    function addSingleLine(e,i) {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (8)](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/859032400/issues/source?bid=2452379&fileBranchId=2815005#l228)